pycbc/waveform/waveform.py

```diff
@@ -769,12 +770,120 @@ def get_td_waveform_from_fd(rwrap=0.2, **params):
     hp = hp.cyclic_time_shift(-rwrap)
     hc = hc.cyclic_time_shift(-rwrap)
 
-    hp = wfutils.fd_to_td(hp, left_window=(nparams['f_lower'],
-                                           params['f_lower']))
-    hc = wfutils.fd_to_td(hc, left_window=(nparams['f_lower'],
-                                           params['f_lower']))
+    hp = wfutils.fd_to_td(hp, delta_t=params['delta_t'],
+                          left_window=(nparams['f_lower'], params['f_lower']))
+    hc = wfutils.fd_to_td(hc, delta_t=params['delta_t'],
+                          left_window=(nparams['f_lower'], params['f_lower']))
     return hp, hc
 
+def get_fd_det_waveform(template=None, **kwargs):
+    """Return a frequency domain gravitational waveform.
+    The waveform generator includes detector response.
+
+    Parameters
+    ----------
+    template: object
+        An object that has attached properties. This can be used to substitute
+        for keyword arguments. An example would be a row in an xml table.
+    {params}
+
+    Returns
+    -------
+    dict
+        The detector-frame waveform (with detector response) in frequency
+        domain. Keys are requested data channels, values are FrequencySeries.
+    """
+    input_params = props(template, **kwargs)
+    input_params['delta_f'] = -1
+    input_params['f_lower'] = -1
+    if input_params['approximant'] not in fd_det:
+        raise ValueError("Approximant %s not available" %
+                            (input_params['approximant']))
+    wav_gen = fd_det[input_params['approximant']]
+    if hasattr(wav_gen, 'required'):
+        required = wav_gen.required
+    else:
+        required = parameters.fd_required
+    check_args(input_params, required)
+    return wav_gen(**input_params)
+
+get_fd_det_waveform.__doc__ = get_fd_det_waveform.__doc__.format(
+    params=parameters.fd_waveform_params.docstr(prefix="    ",
+           include_label=False).lstrip(' '))
+
+def get_td_det_waveform_from_fd_det(template=None, rwrap=0.2, **params):
```

Comment:
Does the implementation need to be duplicated from get_td_waveform_from_fd? Can't you share most of the code?